sequence_processing_pipeline/HumanFilterJob.py

```diff
@@ -0,0 +1,292 @@
+                yield some_list[i:i + n]
+
+        # put these split files in the same location as the fastq_files for
+        # the project. Assume all filepaths in fastq_files have the same
+        # result for dirname().
+        destination_path = dirname(fastq_files[0])
+
+        new_files = []
+        count = 0
+        for chunk in _chunk_list(fastq_files, split_count):
+            trim_file_name = '%s%d' % (self.trim_file, count)
+            trim_file_path = join(destination_path, trim_file_name)
+            with open(trim_file_path, 'w') as f:
+                for line in chunk:
+                    f.write("%s\n" % line)
+            new_files.append(trim_file_path)
+            count += 1
+
+        return new_files
+
+    def _clear_trim_files(self):
+        # remove all files with a name beginning in self.trim_file.
+        # assume cleaning the entire root_dir is overkill, but won't
+        # hurt anything.
+        for root, dirs, files in walk(self.root_dir):
+            for some_file in files:
+                if self.trim_file in some_file:
+                    some_path = join(root, some_file)
+                    remove(some_path)
+
+    def _process_sample_sheet(self, sample_sheet_path):
+        # Some column headers in some sample sheets appear to use
+        # different names for the same metadata. This dict maps the names
+        # commonly found to the internal name used by the code.
+        # TODO: Not certain BarcodesAreRC == PolyGTrimming == a_trim. Confirm
+        name_map = {'Sample_Project': 'project_name', 'Project': 'project_name', 'QiitaID': 'qiita_proj',
+                    'BarcodesAreRC': 'a_trim', 'PolyGTrimming': 'a_trim', 'ForwardAdapter': 'adapter_a',
+                    'ReverseAdapter': 'adapter_A', 'HumanFiltering': 'h_filter'}
+
+        sheet = KLSampleSheet(sample_sheet_path)
+        valid_sheet = validate_sample_sheet(sheet)
+
+        if not valid_sheet:
+            s = "Sample sheet %s is not valid." % sample_sheet_path
+            raise PipelineError(s)
+
+        header = valid_sheet.Header
+        chemistry = header['chemistry']
+
+        bioinformatics = valid_sheet.Bioinformatics
+
+        # reorganize the data into a list of dictionaries, one for each row.
+        # the ordering of the rows will be preserved in the order of the list.
+
+        l = []
+        for i in range(0, len(bioinformatics)):
+            l.append({})
+
+        for item in bioinformatics:
+            my_series = bioinformatics[item]
+            for index, value in my_series.items():
+                key = name_map[item]
+                # convert all manner of positive/negative implying strings
+                # into a true boolean value.
+                if key in ['a_trim', 'h_filter']:
+                    value = value.strip().lower()
+                    value = True if value in ['true', 'yes'] else False
+                l[index][key] = value
```

Comment:
Suggest using Pandas' methods for converting to dict and casting the types. Maybe something like sheet.Bioinformatics.to_dict() and infer_types along the way.